File: admin/import-panel.php
<?php
/**
 * Admin - Import des panélistes depuis Excel
 * Utilise PhpSpreadsheet si disponible, sinon parsing CSV
 */

require_once __DIR__ . '/../api/db.php';

// Vérification session admin (simplifiée pour l'exemple)
session_start();
// TODO: Ajouter vérification authentification admin

$message = '';
$imported = 0;
$errors = [];

// Durée avant expiration (30 jours)
define('EXPIRATION_DAYS', 30);

/**
 * Parse une date JJ/MM/AAAA ou composants séparés
 */
function parseDate($jour, $mois, $annee) {
    if (!$jour || !$mois || !$annee) return null;

    $jour = intval($jour);
    $mois = intval($mois);
    $annee = intval($annee);

    // Correction année sur 2 chiffres
    if ($annee < 100) {
        $annee = ($annee > 25) ? 1900 + $annee : 2000 + $annee;
    }

    if ($jour < 1 || $jour > 31 || $mois < 1 || $mois > 12) return null;

    return sprintf('%04d-%02d-%02d', $annee, $mois, $jour);
}

/**
 * Nettoie un numéro de téléphone
 */
function cleanPhone($phone) {
    if (!$phone) return null;
    $phone = preg_replace('/[^0-9]/', '', $phone);
    if (strlen($phone) === 9) {
        $phone = '0' . $phone;
    }
    return $phone ?: null;
}

/**
 * Convertit Oui/Non en boolean
 */
function parseBool($value) {
    if ($value === null) return null;
    $val = strtolower(trim($value));
    return in_array($val, ['oui', 'yes', '1', 'true']) ? 1 : 0;
}

/**
 * Import depuis un fichier CSV
 */
function importFromCSV($filepath) {
    global $imported, $errors;

    $handle = fopen($filepath, 'r');
    if (!$handle) {
        throw new Exception("Impossible d'ouvrir le fichier");
    }

    // Lecture des en-têtes
    $headers = fgetcsv($handle, 0, ';');
    if (!$headers) {
        throw new Exception("Fichier vide ou format invalide");
    }

    // Normalise les en-têtes
    $headers = array_map('trim', $headers);
    $headers = array_map('strtoupper', $headers);

    $expiresAt = date('Y-m-d H:i:s', strtotime('+' . EXPIRATION_DAYS . ' days'));
    $lineNum = 1;

    while (($row = fgetcsv($handle, 0, ';')) !== false) {
        $lineNum++;

        try {
            // Crée un tableau associatif
            $data = [];
            foreach ($headers as $i => $header) {
                $data[$header] = isset($row[$i]) ? trim($row[$i]) : null;
            }

            // Champs obligatoires
            $panelId = $data['ID'] ?? null;
            $email = strtolower($data['EMAIL'] ?? '');

            if (!$panelId || !$email || strpos($email, '@') === false) {
                continue; // Skip
            }

            // Extraction des données
            $insertData = [
                'panel_id' => intval($panelId),
                'email' => $email,
                'region' => $data['REGION'] ?? null,
                'civilite' => $data['CIVILITE'] ?? null,
                'nom' => $data['NOM'] ?? null,
                'prenom' => $data['PRENOM'] ?? null,
                'adresse' => $data['ADRESSE_POSTAL'] ?? null,
                'code_postal' => $data['CODE_POSTAL'] ?? null,
                'departement' => $data['DEPARTEMENT'] ?? null,
                'ville' => $data['VILLE'] ?? null,
                'tel_domicile' => cleanPhone($data['TEL_DOMICILE'] ?? null),
                'tel_portable' => cleanPhone($data['TEL_PORTABLE'] ?? null),
                'tel_bureau' => cleanPhone($data['TEL_BUREAU'] ?? null),
                'date_naissance' => parseDate($data['JJ'] ?? null, $data['MM'] ?? null, $data['AA'] ?? null),
                'age' => isset($data['AGE']) ? intval($data['AGE']) : null,
                'situation_familiale' => $data['SITUATION_FAMILIAL'] ?? null,
                'situation_professionnelle' => $data['SITUATION_PROFESSIONNEL'] ?? null,
                'enfants_au_foyer' => intval($data['ENFANTS_AU_FOYER'] ?? 0),
                'enfants_plus_18ans' => intval($data['ENFANTS_PLUS_DE_18ANS'] ?? 0),
                'enfants_moins_18ans' => intval($data['ENFANTS_MOINS_DE_18ANS'] ?? 0),
                'diplome' => $data['DIPLOME_OBTENU'] ?? null,
                'profession' => $data['PROFESSION_ACTUEL'] ?? null,
                'secteur_activite' => $data['SECTEUR_ACTIVITE'] ?? null,
                'revenu_mensuel' => $data['REVENU_NET_MENSUEL_DU_FOYER'] ?? null,
                'banque_principale' => $data['BANQUE_PRINCIPAL'] ?? null,
                'autres_banques' => $data['AUTRES_BANQUES'] ?? null,
                'type_habitation' => $data['TYPE_HABITATION'] ?? null,
                'situation_habitation' => $data['SITUATION_HABITATION'] ?? null,
                'possede_voiture' => parseBool($data['VOITURE'] ?? null),
                'possede_moto' => parseBool($data['MOTO'] ?? null),
                'expires_at' => $expiresAt
            ];

            // Équipements
            $equipements = [];
            $equipCols = [
                'POSSEDE_TELEVISEUR' => 'televiseur',
                'POSSEDE_VIDEO_PROJECTEUR' => 'video_projecteur',
                'POSSEDE_UN_ORDINATEUR_PORTABLE' => 'ordinateur_portable',
                'POSSEDE_APPAREIL_PHOTO_NUMERIQUE' => 'appareil_photo',
                'POSSEDE_WIFI' => 'wifi',
                'POSSEDE_GPS' => 'gps'
            ];
            foreach ($equipCols as $col => $key) {
                if (isset($data[$col]) && parseBool($data[$col])) {
                    $equipements[$key] = true;
                }
            }
            $insertData['equipements'] = !empty($equipements) ? json_encode($equipements) : null;

            // Conjoint
            if (!empty($data['CIVILITE_CONJOINT'])) {
                $conjoint = [
                    'civilite' => $data['CIVILITE_CONJOINT'] ?? null,
                    'nom' => $data['NOM_CONJOINT'] ?? null,
                    'prenom' => $data['PRENOM_CONJOINT'] ?? null,
                    'email' => $data['EMAIL_CONJOINT'] ?? null,
                    'date_naissance' => parseDate($data['JJ_CONJOINT'] ?? null, $data['MM_CONJOINT'] ?? null, $data['AA_CONJOINT'] ?? null),
                    'diplome' => $data['DIPLOME_OBTENU_CONJOINT'] ?? null,
                    'profession' => $data['PROFESSION_ACTUEl_CONJOINT'] ?? null
                ];
                $insertData['conjoint_data'] = json_encode($conjoint);
            } else {
                $insertData['conjoint_data'] = null;
            }

            // Insert ou update
            $sql = "INSERT INTO panel_imported (
                panel_id, email, region, civilite, nom, prenom,
                adresse, code_postal, departement, ville,
                tel_domicile, tel_portable, tel_bureau,
                date_naissance, age, situation_familiale, situation_professionnelle,
                enfants_au_foyer, enfants_plus_18ans, enfants_moins_18ans,
                diplome, profession, secteur_activite, revenu_mensuel,
                banque_principale, autres_banques, conjoint_data,
                type_habitation, situation_habitation,
                possede_voiture, possede_moto, equipements, expires_at
            ) VALUES (
                :panel_id, :email, :region, :civilite, :nom, :prenom,
                :adresse, :code_postal, :departement, :ville,
                :tel_domicile, :tel_portable, :tel_bureau,
                :date_naissance, :age, :situation_familiale, :situation_professionnelle,
                :enfants_au_foyer, :enfants_plus_18ans, :enfants_moins_18ans,
                :diplome, :profession, :secteur_activite, :revenu_mensuel,
                :banque_principale, :autres_banques, :conjoint_data,
                :type_habitation, :situation_habitation,
                :possede_voiture, :possede_moto, :equipements, :expires_at
            ) ON DUPLICATE KEY UPDATE
                nom = VALUES(nom),
                prenom = VALUES(prenom),
                expires_at = VALUES(expires_at)";

            dbExecute($sql, $insertData);
            $imported++;

        } catch (Exception $e) {
            $errors[] = "Ligne $lineNum: " . $e->getMessage();
            if (count($errors) > 50) break; // Limite les erreurs affichées
        }
    }

    fclose($handle);
}

// Traitement de l'upload
if ($_SERVER['REQUEST_METHOD'] === 'POST' && isset($_FILES['panel_file'])) {
    $file = $_FILES['panel_file'];

    if ($file['error'] !== UPLOAD_ERR_OK) {
        $message = 'Erreur lors de l\'upload du fichier.';
    } else {
        $extension = strtolower(pathinfo($file['name'], PATHINFO_EXTENSION));

        if ($extension === 'csv') {
            try {
                importFromCSV($file['tmp_name']);
                $message = "Import terminé: $imported enregistrements importés.";
                if (!empty($errors)) {
                    $message .= " " . count($errors) . " erreurs.";
                }
            } catch (Exception $e) {
                $message = 'Erreur: ' . $e->getMessage();
            }
        } elseif ($extension === 'xlsx') {
            $message = 'Pour les fichiers Excel, veuillez d\'abord les convertir en CSV (séparateur: point-virgule).';
        } else {
            $message = 'Format non supporté. Utilisez CSV ou XLSX.';
        }
    }
}

// Statistiques
$stats = dbQueryOne("SELECT
    COUNT(*) as total,
    SUM(CASE WHEN claimed = 1 THEN 1 ELSE 0 END) as claimed,
    SUM(CASE WHEN claimed = 0 AND (expires_at IS NULL OR expires_at > NOW()) THEN 1 ELSE 0 END) as available,
    SUM(CASE WHEN expires_at < NOW() AND claimed = 0 THEN 1 ELSE 0 END) as expired
FROM panel_imported");
?>
<!DOCTYPE html>
<html lang="fr">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Import Panel - Admin</title>
    <script src="https://cdn.tailwindcss.com"></script>
</head>
<body class="bg-gray-100 min-h-screen">
    <div class="container mx-auto px-4 py-8 max-w-4xl">
        <h1 class="text-3xl font-bold text-gray-800 mb-8">Import des Panélistes</h1>

        <?php if ($message): ?>
            <div class="mb-6 p-4 rounded-lg <?= strpos($message, 'Erreur') !== false ? 'bg-red-100 text-red-700' : 'bg-green-100 text-green-700' ?>">
                <?= htmlspecialchars($message) ?>
            </div>
        <?php endif; ?>

        <?php if (!empty($errors)): ?>
            <div class="mb-6 p-4 bg-yellow-100 text-yellow-700 rounded-lg">
                <strong>Erreurs détectées:</strong>
                <ul class="list-disc ml-5 mt-2">
                    <?php foreach (array_slice($errors, 0, 10) as $error): ?>
                        <li><?= htmlspecialchars($error) ?></li>
                    <?php endforeach; ?>
                </ul>
                <?php if (count($errors) > 10): ?>
                    <p class="mt-2 italic">... et <?= count($errors) - 10 ?> autres erreurs</p>
                <?php endif; ?>
            </div>
        <?php endif; ?>

        <!-- Statistiques -->
        <div class="grid grid-cols-4 gap-4 mb-8">
            <div class="bg-white rounded-lg shadow p-4 text-center">
                <div class="text-3xl font-bold text-blue-600"><?= number_format($stats['total'] ?? 0) ?></div>
                <div class="text-gray-500 text-sm">Total importés</div>
            </div>
            <div class="bg-white rounded-lg shadow p-4 text-center">
                <div class="text-3xl font-bold text-green-600"><?= number_format($stats['available'] ?? 0) ?></div>
                <div class="text-gray-500 text-sm">Disponibles</div>
            </div>
            <div class="bg-white rounded-lg shadow p-4 text-center">
                <div class="text-3xl font-bold text-purple-600"><?= number_format($stats['claimed'] ?? 0) ?></div>
                <div class="text-gray-500 text-sm">Réclamés</div>
            </div>
            <div class="bg-white rounded-lg shadow p-4 text-center">
                <div class="text-3xl font-bold text-gray-400"><?= number_format($stats['expired'] ?? 0) ?></div>
                <div class="text-gray-500 text-sm">Expirés</div>
            </div>
        </div>

        <!-- Formulaire d'import -->
        <div class="bg-white rounded-lg shadow p-6 mb-8">
            <h2 class="text-xl font-semibold mb-4">Importer un fichier</h2>

            <form method="POST" enctype="multipart/form-data" class="space-y-4">
                <div>
                    <label class="block text-gray-700 mb-2">Fichier CSV (séparateur: point-virgule)</label>
                    <input type="file" name="panel_file" accept=".csv,.xlsx" required
                           class="block w-full text-gray-500 file:mr-4 file:py-2 file:px-4
                                  file:rounded-full file:border-0 file:text-sm file:font-semibold
                                  file:bg-blue-50 file:text-blue-700 hover:file:bg-blue-100">
                </div>

                <div class="bg-yellow-50 border border-yellow-200 rounded p-4 text-sm text-yellow-700">
                    <strong>Note:</strong> Pour les fichiers Excel (.xlsx), exportez d'abord en CSV avec le séparateur point-virgule.
                    <br>Les données expireront automatiquement après 30 jours si non réclamées.
                </div>

                <button type="submit"
                        class="bg-blue-600 text-white px-6 py-2 rounded-lg hover:bg-blue-700 transition">
                    Importer
                </button>
            </form>
        </div>

        <!-- Instructions -->
        <div class="bg-white rounded-lg shadow p-6">
            <h2 class="text-xl font-semibold mb-4">Comment ça fonctionne</h2>

            <div class="space-y-3 text-gray-600">
                <p><strong>1.</strong> Importez votre fichier panel (CSV avec colonnes: ID, EMAIL, NOM, PRENOM, etc.)</p>
                <p><strong>2.</strong> Les données sont stockées avec une date d'expiration de 30 jours</p>
                <p><strong>3.</strong> Quand un utilisateur s'inscrit sur l'app, son email est vérifié</p>
                <p><strong>4.</strong> S'il correspond, ses données sont pré-remplies et il n'a qu'à vérifier</p>
                <p><strong>5.</strong> Après 30 jours, les données non réclamées sont automatiquement supprimées</p>
            </div>
        </div>

        <!-- Bouton nettoyage manuel -->
        <div class="mt-6 text-center">
            <a href="?cleanup=1" class="text-red-600 hover:text-red-800 text-sm"
               onclick="return confirm('Supprimer tous les enregistrements expirés ?')">
                Nettoyer les enregistrements expirés
            </a>
        </div>

        <?php if (isset($_GET['cleanup'])): ?>
            <?php
            $deleted = dbExecute("DELETE FROM panel_imported WHERE claimed = 0 AND expires_at < NOW()");
            ?>
            <div class="mt-4 p-4 bg-green-100 text-green-700 rounded-lg text-center">
                Nettoyage effectué.
            </div>
        <?php endif; ?>
    </div>
</body>
</html>

File: scripts/import_panel.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
"""
Script d'import des panélistes Excel vers MySQL
Usage: python import_panel.py [fichier.xlsx]
"""

import sys
import json
import mysql.connector
from datetime import datetime, timedelta
from openpyxl import load_workbook

# Configuration base de données (à adapter selon votre environnement)
DB_CONFIG = {
    'host': 'localhost',
    'user': 'root',
    'password': '',
    'database': 'u486755141_etudes',
    'charset': 'utf8mb4'
}

# Durée avant expiration (1 mois)
EXPIRATION_DAYS = 30


def parse_date(jour, mois, annee):
    """Convertit JJ/MM/AA en date"""
    try:
        if jour and mois and annee:
            jour = int(jour)
            mois = int(mois)
            annee = int(annee)
            # Correction année sur 2 chiffres
            if annee < 100:
                annee = 1900 + annee if annee > 25 else 2000 + annee
            return f"{annee:04d}-{mois:02d}-{jour:02d}"
    except:
        pass
    return None


def parse_bool(value):
    """Convertit Oui/Non en boolean"""
    if value is None:
        return None
    val = str(value).strip().lower()
    return 1 if val in ['oui', 'yes', '1', 'true'] else 0


def clean_phone(phone):
    """Nettoie les numéros de téléphone"""
    if phone is None:
        return None
    phone = str(phone).strip()
    if phone in ['', 'None', '0']:
        return None
    # Ajoute le 0 devant si nécessaire
    if len(phone) == 9 and phone.isdigit():
        phone = '0' + phone
    return phone


def extract_children_data(row, headers):
    """Extrait les données des enfants"""
    children = []
    for i in range(1, 7):
        sexe_col = f'SEXE_ENFANT{i}'
        if sexe_col in headers:
            idx = headers.index(sexe_col)
            sexe = row[idx] if idx < len(row) else None
            if sexe:
                child = {
                    'sexe': sexe,
                    'date_naissance': parse_date(
                        row[headers.index(f'JJ_ENFANT{i}')] if f'JJ_ENFANT{i}' in headers else None,
                        row[headers.index(f'MM_ENFANT{i}')] if f'MM_ENFANT{i}' in headers else None,
                        row[headers.index(f'AA_ENFANT{i}')] if f'AA_ENFANT{i}' in headers else None
                    ),
                    'nom': row[headers.index(f'NOM_ENFANT{i}')] if f'NOM_ENFANT{i}' in headers else None,
                    'prenom': row[headers.index(f'PRENOM_ENFANT{i}')] if f'PRENOM_ENFANT{i}' in headers else None,
                    'accepte_test': parse_bool(row[headers.index(f'ACCEPT_IL_DE_PARTICIPER_AU_TEST_ENFANT{i}')] if f'ACCEPT_IL_DE_PARTICIPER_AU_TEST_ENFANT{i}' in headers else None)
                }
                children.append(child)
    return children if children else None


def extract_conjoint_data(row, headers):
    """Extrait les données du conjoint"""
    civilite = row[headers.index('CIVILITE_CONJOINT')] if 'CIVILITE_CONJOINT' in headers else None
    if not civilite:
        return None

    return {
        'civilite': civilite,
        'nom': row[headers.index('NOM_CONJOINT')] if 'NOM_CONJOINT' in headers else None,
        'prenom': row[headers.index('PRENOM_CONJOINT')] if 'PRENOM_CONJOINT' in headers else None,
        'tel_domicile': clean_phone(row[headers.index('TEL_DOMICILE_CONJOINT')] if 'TEL_DOMICILE_CONJOINT' in headers else None),
        'tel_portable': clean_phone(row[headers.index('TEL_PORTABLE_CONJOINT')] if 'TEL_PORTABLE_CONJOINT' in headers else None),
        'tel_bureau': clean_phone(row[headers.index('TEL_BUREAU_CONJOINT')] if 'TEL_BUREAU_CONJOINT' in headers else None),
        'email': row[headers.index('EMAIL_CONJOINT')] if 'EMAIL_CONJOINT' in headers else None,
        'date_naissance': parse_date(
            row[headers.index('JJ_CONJOINT')] if 'JJ_CONJOINT' in headers else None,
            row[headers.index('MM_CONJOINT')] if 'MM_CONJOINT' in headers else None,
            row[headers.index('AA_CONJOINT')] if 'AA_CONJOINT' in headers else None
        ),
        'diplome': row[headers.index('DIPLOME_OBTENU_CONJOINT')] if 'DIPLOME_OBTENU_CONJOINT' in headers else None,
        'situation_professionnelle': row[headers.index('SITUATION_PROFESSIONNEL_CONJOINT')] if 'SITUATION_PROFESSIONNEL_CONJOINT' in headers else None,
        'profession': row[headers.index('PROFESSION_ACTUEl_CONJOINT')] if 'PROFESSION_ACTUEl_CONJOINT' in headers else None,
        'accepte_test': parse_bool(row[headers.index('ACCEPT_IL_DE_PARTICIPER_AU_TEST')] if 'ACCEPT_IL_DE_PARTICIPER_AU_TEST' in headers else None),
        'banque_principale': row[headers.index('BANQUE_PRINCIPAL_CONJOINT')] if 'BANQUE_PRINCIPAL_CONJOINT' in headers else None,
        'autres_banques': row[headers.index('AUTRES_BANQUES_CONJOINT')] if 'AUTRES_BANQUES_CONJOINT' in headers else None,
    }


def extract_vehicles_data(row, headers, prefix='VOITURE'):
    """Extrait les données des véhicules"""
    vehicles = []
    for i in range(1, 4):
        modele_col = f'MODELE_{prefix}' if i == 1 else f'MODELE_{prefix}{i}'
        if modele_col in headers:
            idx = headers.index(modele_col)
            modele = row[idx] if idx < len(row) else None
            if modele and str(modele).strip():
                vehicle = {
                    'modele': modele,
                    'annee_achat': row[headers.index(f'ANNEE_ACHAT_{prefix}{i}')] if f'ANNEE_ACHAT_{prefix}{i}' in headers else None,
                    'type': row[headers.index(f'TYPE_DE_VEHICULE_{prefix}{i}')] if f'TYPE_DE_VEHICULE_{prefix}{i}' in headers else None,
                    'type_achat': row[headers.index(f'TYPE_ACHAT_{prefix}{i}')] if f'TYPE_ACHAT_{prefix}{i}' in headers else None,
                }
                vehicles.append(vehicle)
    return vehicles if vehicles else None


def extract_equipements(row, headers):
    """Extrait les équipements"""
    equipements = {}
    mapping = {
        'POSSEDE_TELEVISEUR': 'televiseur',
        'POSSEDE_VIDEO_PROJECTEUR': 'video_projecteur',
        'POSSEDE_UN_ORDINATEUR_PORTABLE': 'ordinateur_portable',
        'POSSEDE_APPAREIL_PHOTO_NUMERIQUE': 'appareil_photo',
        'POSSEDE_WIFI': 'wifi',
        'POSSEDE_GPS': 'gps'
    }
    for col, key in mapping.items():
        if col in headers:
            val = row[headers.index(col)]
            if val:
                equipements[key] = parse_bool(val)
    return equipements if equipements else None


def import_excel_to_mysql(excel_file):
    """Importe le fichier Excel dans MySQL"""
    print(f"Chargement de {excel_file}...")
    wb = load_workbook(excel_file, read_only=True)
    ws = wb.active

    # Récupère les en-têtes
    headers = []
    for cell in ws[1]:
        headers.append(cell.value if cell.value else f'COL_{len(headers)}')

    print(f"Colonnes: {len(headers)}")
    print(f"Lignes: {ws.max_row - 1}")

    # Connexion MySQL
    print("Connexion à MySQL...")
    conn = mysql.connector.connect(**DB_CONFIG)
    cursor = conn.cursor()

    # Date d'expiration (1 mois après aujourd'hui)
    expires_at = (datetime.now() + timedelta(days=EXPIRATION_DAYS)).strftime('%Y-%m-%d %H:%M:%S')

    # Préparation de l'insertion
    insert_sql = """
        INSERT INTO panel_imported (
            panel_id, email, region, civilite, nom, prenom,
            adresse, code_postal, departement, ville,
            tel_domicile, tel_portable, tel_bureau,
            date_naissance, age, situation_familiale, situation_professionnelle,
            enfants_au_foyer, enfants_plus_18ans, enfants_moins_18ans, enfants_data,
            diplome, profession, secteur_activite, revenu_mensuel,
            banque_principale, autres_banques, conjoint_data,
            type_habitation, situation_habitation,
            possede_voiture, voitures_data, possede_moto, motos_data,
            equipements, expires_at
        ) VALUES (
            %s, %s, %s, %s, %s, %s,
            %s, %s, %s, %s,
            %s, %s, %s,
            %s, %s, %s, %s,
            %s, %s, %s, %s,
            %s, %s, %s, %s,
            %s, %s, %s,
            %s, %s,
            %s, %s, %s, %s,
            %s, %s
        ) ON DUPLICATE KEY UPDATE
            region = VALUES(region),
            civilite = VALUES(civilite),
            nom = VALUES(nom),
            prenom = VALUES(prenom),
            expires_at = VALUES(expires_at)
    """

    imported = 0
    skipped = 0
    errors = 0

    print("Import en cours...")

    for row_num, row in enumerate(ws.iter_rows(min_row=2, values_only=True), start=2):
        try:
            # Vérifie les champs obligatoires
            panel_id = row[headers.index('ID')] if 'ID' in headers else None
            email = row[headers.index('EMAIL')] if 'EMAIL' in headers else None

            if not panel_id or not email:
                skipped += 1
                continue

            # Nettoyage email
            email = str(email).strip().lower()
            if '@' not in email:
                skipped += 1
                continue

            # Extraction des données
            data = (
                int(panel_id),
                email,
                row[headers.index('REGION')] if 'REGION' in headers else None,
                row[headers.index('CIVILITE')] if 'CIVILITE' in headers else None,
                row[headers.index('NOM')] if 'NOM' in headers else None,
                row[headers.index('PRENOM')] if 'PRENOM' in headers else None,
                row[headers.index('ADRESSE_POSTAL')] if 'ADRESSE_POSTAL' in headers else None,
                str(row[headers.index('CODE_POSTAL')]) if 'CODE_POSTAL' in headers and row[headers.index('CODE_POSTAL')] else None,
                str(row[headers.index('DEPARTEMENT')]) if 'DEPARTEMENT' in headers and row[headers.index('DEPARTEMENT')] else None,
                row[headers.index('VILLE')] if 'VILLE' in headers else None,
                clean_phone(row[headers.index('TEL_DOMICILE')] if 'TEL_DOMICILE' in headers else None),
                clean_phone(row[headers.index('TEL_PORTABLE')] if 'TEL_PORTABLE' in headers else None),
                clean_phone(row[headers.index('TEL_BUREAU')] if 'TEL_BUREAU' in headers else None),
                parse_date(
                    row[headers.index('JJ')] if 'JJ' in headers else None,
                    row[headers.index('MM')] if 'MM' in headers else None,
                    row[headers.index('AA')] if 'AA' in headers else None
                ),
                int(row[headers.index('AGE')]) if 'AGE' in headers and row[headers.index('AGE')] else None,
                row[headers.index('SITUATION_FAMILIAL')] if 'SITUATION_FAMILIAL' in headers else None,
                row[headers.index('SITUATION_PROFESSIONNEL')] if 'SITUATION_PROFESSIONNEL' in headers else None,
                int(row[headers.index('ENFANTS_AU_FOYER')] or 0) if 'ENFANTS_AU_FOYER' in headers else 0,
                int(row[headers.index('ENFANTS_PLUS_DE_18ANS')] or 0) if 'ENFANTS_PLUS_DE_18ANS' in headers else 0,
                int(row[headers.index('ENFANTS_MOINS_DE_18ANS')] or 0) if 'ENFANTS_MOINS_DE_18ANS' in headers else 0,
                json.dumps(extract_children_data(row, headers), ensure_ascii=False) if extract_children_data(row, headers) else None,
                row[headers.index('DIPLOME_OBTENU')] if 'DIPLOME_OBTENU' in headers else None,
                row[headers.index('PROFESSION_ACTUEL')] if 'PROFESSION_ACTUEL' in headers else None,
                row[headers.index('SECTEUR_ACTIVITE')] if 'SECTEUR_ACTIVITE' in headers else None,
                row[headers.index('REVENU_NET_MENSUEL_DU_FOYER')] if 'REVENU_NET_MENSUEL_DU_FOYER' in headers else None,
                row[headers.index('BANQUE_PRINCIPAL')] if 'BANQUE_PRINCIPAL' in headers else None,
                row[headers.index('AUTRES_BANQUES')] if 'AUTRES_BANQUES' in headers else None,
                json.dumps(extract_conjoint_data(row, headers), ensure_ascii=False) if extract_conjoint_data(row, headers) else None,
                row[headers.index('TYPE_HABITATION')] if 'TYPE_HABITATION' in headers else None,
                row[headers.index('SITUATION_HABITATION')] if 'SITUATION_HABITATION' in headers else None,
                parse_bool(row[headers.index('VOITURE')] if 'VOITURE' in headers else None),
                json.dumps(extract_vehicles_data(row, headers, 'VOITURE'), ensure_ascii=False) if extract_vehicles_data(row, headers, 'VOITURE') else None,
                parse_bool(row[headers.index('MOTO')] if 'MOTO' in headers else None),
                json.dumps(extract_vehicles_data(row, headers, 'MOTO'), ensure_ascii=False) if extract_vehicles_data(row, headers, 'MOTO') else None,
                json.dumps(extract_equipements(row, headers), ensure_ascii=False) if extract_equipements(row, headers) else None,
                expires_at
            )

            cursor.execute(insert_sql, data)
            imported += 1

            if imported % 1000 == 0:
                conn.commit()
                print(f"  {imported} enregistrements importés...")

        except Exception as e:
            errors += 1
            if errors <= 10:
                print(f"  Erreur ligne {row_num}: {e}")

    conn.commit()
    cursor.close()
    conn.close()

    print("\n=== RÉSUMÉ ===")
    print(f"Importés: {imported}")
    print(f"Ignorés (sans email/ID): {skipped}")
    print(f"Erreurs: {errors}")
    print(f"Expiration: {expires_at}")


if __name__ == '__main__':
    if len(sys.argv) < 2:
        excel_file = 'panel_clean.xlsx'
    else:
        excel_file = sys.argv[1]

    import_excel_to_mysql(excel_file)
